tests/misc/test_extra_group_pkgs_can_be_installed.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import pytest
+
+from lib.commands import SSHCommandFailed
+from packaging import version
+from subprocess import run, CalledProcessError
+
+def install_failed(pkg_name, h):
+    """Check if a package can be installed or not.
+
+    Returns True if the package cannot be installed, False otherwise.
+    It saves the yum state before trying to install a package and restore
+    it after.
+    """
+    failed = False
+    h.yum_save_state()
+
+    try:
+        h.yum_install([pkg_name])
+    except SSHCommandFailed:
+        failed = True
+
+    h.yum_restore_saved_state()
+    return failed
+
+def test_extra_group_packages_can_be_installed(host):
+    v = version.parse(host.inventory['PRODUCT_VERSION'])
+    url = f"https://reports.xcp-ng.org/{v}/extra_installable.txt"
+    logging.info(f"getting extra packages from {url}")
+
+    try:
+        ret = run(
+            ["wget", "-O-", "-q", url],
+            capture_output=True,
+            check=True)
+    except CalledProcessError:
+        logging.error(f"failed to get extra installable list from {url}")
+        assert False
+
+    pkgs = ret.stdout.decode("utf-8").splitlines()
+    logging.info(f"{pkgs}")
+    failures= [p for p in pkgs if install_failed(p, host)]
+
+    assert failures == []
```

Comment:
You don't need to store failures and assert at the end. Instead, just let the yum command raise an error, and pytest will convert this in a test failure. Which means of course that you must ensure the teardown part is always executed to restore the previous state, but this is easily done through pytest's `fixtures`.

We would lose some information, as not all packages would be tested, but we usually stop at the first error when we run our testsuites anyway, because a failed test may leave the system under test in a state which is not suitable for the next tests, despite our efforts at restoring a good state. Or an early test failure may reveal that we launched tests on a system that is not in an appropriate state.

Reply:
I think that with a list of packages that failed we are able to test all packages and the ouptut will be easy to read. Let the yum command raises an error will be less readable I think. But both are ok for me.